%YAML 1.2
---
name: global
components:
- name: adapter
  type: nvidia::isaac::tensor_ops::ImageAdapter
  parameters:
    message_type: "VideoBuffer"
- name: stream
  type: nvidia::isaac::tensor_ops::TensorStream
  parameters:
    backend_type: "VPI"
    engine_type: "GPU"
- name: encoder_context
  type: nvidia::gxf::VideoEncoderContext
  parameters:
    scheduling_term: encoder_response/async_st
---
name: encoder
components:
- name: data_receiver
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 1
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: data_receiver
    min_size: 1
- name: encoder_request
  type: nvidia::gxf::VideoEncoderRequest
  parameters:
    input_frame: data_receiver
    inbuf_storage_type: 1
    profile: 0
    qp: 22
    hw_preset_type: 1
    input_width: 1920
    input_height: 1200
    input_format: nv12
    config: pframe_cqp
    iframe_interval: 5
    rate_control_mode: 0
    level: 14
    videoencoder_context: global/encoder_context
---
name: encoder_response
components:
- name: output
  type: nvidia::gxf::DoubleBufferTransmitter
  parameters:
    capacity: 1
- type: nvidia::gxf::DownstreamReceptiveSchedulingTerm
  parameters:
    transmitter: output
    min_size: 1
- name: pool
  type: nvidia::gxf::BlockMemoryPool
  parameters:
    storage_type: 0
    block_size: 15000000
    num_blocks: 8
- type: nvidia::gxf::VideoEncoderResponse
  parameters:
    videoencoder_context: global/encoder_context
    output_transmitter: output
    outbuf_storage_type: 0
    pool: pool
- name: async_st
  type: nvidia::gxf::AsynchronousSchedulingTerm
- type: nvidia::gxf::MemoryAvailableSchedulingTerm
  parameters:
    allocator: pool
    min_blocks: 1
---
name: sink
components:
- name: signal
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 1
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: signal
    min_size: 1
- name: sink
  type: nvidia::isaac_ros::MessageRelay
  parameters:
    source: signal
    max_waiting_count: 1
    drop_waiting: false
---
components:
- name: clock
  type: nvidia::gxf::RealtimeClock
- type: nvidia::gxf::EventBasedScheduler
  parameters:
    clock: clock
    stop_on_deadlock: false
- type: nvidia::gxf::JobStatistics
  parameters:
    clock: clock
- type: nvidia::gxf::Connection
  parameters:
    source: encoder_response/output
    target: sink/signal